test/integration/sync.test.js

```diff
@@ -0,0 +1,363 @@
+runner.run(testFunc);
+
+function testFunc() {
+
+  const dataStoreTypes = [Kinvey.DataStoreType.Cache, Kinvey.DataStoreType.Sync];
+  let networkStore;
+  let syncStore;
+  let cacheStore;
+  let storeToTest;
+  const notFoundErrorName = 'NotFoundError';
+
+  //validates Push operation result for 1 created, 1 modified and 1 deleted locally items
+  const validatePushOperation = (result, createdItem, modifiedItem, deletedItem, expectedServerItemsCount) => {
+    return new Promise((resolve, reject) => {
+      expect(result.length).to.equal(3);
```

Comment:
There is no synchronicity here - this doesn't need to be inside the Promise constructor. And returning `networkStore.find().toPromise()` is enough - no need for the Promise constructor at all.